modules/eslint-plugin/src/rules/signals/signal-state-no-arrays-at-root-level.ts

```diff
@@ -8,17 +8,33 @@ export const messageId = 'signalStateNoArraysAtRootLevel';
 type MessageIds = typeof messageId;
 type Options = readonly [];
 
+const NON_RECORD_TYPES = [
+  'Array',
+  'Set',
+  'Map',
+  'WeakSet',
+  'WeakMap',
+  'Date',
+  'Error',
+  'RegExp',
+  'ArrayBuffer',
+  'DataView',
+  'Promise',
+  'Function',
+];
+
 export default createRule<Options, MessageIds>({
   name: path.parse(__filename).name,
   meta: {
     type: 'problem',
     docs: {
       description: `signalState should accept a record or dictionary as an input argument.`,
       ngrxModule: 'signals',
+      requiresTypeChecking: true,
     },
     schema: [],
     messages: {
-      [messageId]: `Wrap the array in an record or dictionary.`,
+      [messageId]: `signalState should accept a record or dictionary as an input argument.`,
```

Comment:
What about using the following message here?
While reporting the error, the `property` needs to be provided.
For an example see `use-effects-lifecycle-interface`.

```suggestion
      [messageId]: `The property type `{{ property }}` is forbidden as the initial state argument, wrap the property in a record or dictionary.,
```